main/test/ca/uwaterloo/flix/library/TestMap.flix

```diff
@@ -0,0 +1,1289 @@
+
+/////////////////////////////////////////////////////////////////////////////
+// foldLeft                                                                //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def foldLeft01: Bool = Map.foldLeft((s, v) -> s + v, 0, @{}) `assertEq!` 0
+
+@test
+def foldLeft02: Bool = Map.foldLeft((s, v) -> s + v, 0, @{1 -> 2}) `assertEq!` 2
+
+@test
+def foldLeft03: Bool = Map.foldLeft((s, v) -> s + v, 0, @{1 -> 2, 3 -> 4}) `assertEq!` 6
+
+@test
+def foldLeft04: Bool = Map.foldLeft((s, v) -> s + v, 0, @{1 -> 2, 3 -> 4, 5 -> 6}) `assertEq!` 12
+
+/////////////////////////////////////////////////////////////////////////////
+// foldLeftWithKey                                                         //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def foldLeftWithKey01: Bool = Map.foldLeftWithKey((k, s, v) -> k + s + v, 0, @{}) `assertEq!` 0
+
+@test
+def foldLeftWithKey02: Bool = Map.foldLeftWithKey((k, s, v) -> k + s + v, 0, @{1 -> 2}) `assertEq!` 3
+
+@test
+def foldLeftWithKey03: Bool = Map.foldLeftWithKey((k, s, v) -> k + s + v, 0, @{1 -> 2, 3 -> 4}) `assertEq!` 10
+
+@test
+def foldLeftWithKey04: Bool = Map.foldLeftWithKey((k, s, v) -> k + s + v, 0, @{1 -> 2, 3 -> 4, 5 -> 6}) `assertEq!` 21
+
+/////////////////////////////////////////////////////////////////////////////
+// foldRight                                                               //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def foldRight01: Bool = Map.foldRight((v, s) -> s + v, 0, @{}) `assertEq!` 0
+
+@test
+def foldRight02: Bool = Map.foldRight((v, s) -> s + v, 0, @{1 -> 2}) `assertEq!` 2
+
+@test
+def foldRight03: Bool = Map.foldRight((v, s) -> s + v, 0, @{1 -> 2, 3 -> 4}) `assertEq!` 6
+
+@test
+def foldRight04: Bool = Map.foldRight((v, s) -> s + v, 0, @{1 -> 2, 3 -> 4, 5 -> 6}) `assertEq!` 12
+
+/////////////////////////////////////////////////////////////////////////////
+// foldRightWithKey                                                        //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def foldRightWithKey01: Bool = Map.foldRightWithKey((k, v, s) -> k + s + v, 0, @{}) `assertEq!` 0
+
+@test
+def foldRightWithKey02: Bool = Map.foldRightWithKey((k, v, s) -> k + s + v, 0, @{1 -> 2}) `assertEq!` 3
+
+@test
+def foldRightWithKey03: Bool = Map.foldRightWithKey((k, v, s) -> k + s + v, 0, @{1 -> 2, 3 -> 4}) `assertEq!` 10
+
+@test
+def foldRightWithKey04: Bool = Map.foldRightWithKey((k, v, s) -> k + s + v, 0, @{1 -> 2, 3 -> 4, 5 -> 6}) `assertEq!` 21
+
+/////////////////////////////////////////////////////////////////////////////
+// union                                                                   //
+/////////////////////////////////////////////////////////////////////////////
+//@test
+//def union01: Bool = assert!(Map.eq(Map.union(@{}, @{}), @{}))
```

Comment:
Some tests on the empty map create strange errors, I'll be checking my implementations.